### PR TITLE
samples: bluetooth: nrf_dm: Add energy consumption information.

### PR DESCRIPTION
**Pull request by maje-emb**
Adds information to the documentation about the energy consumption of the sample.

### CODE REVIEW

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by peknis** [comment]
Do not forget the changelog.

**Comment by maje-emb** [comment]
> Do not forget the changelog.

Fixed. Added an entry to the changelog.

File: samples/bluetooth/nrf_dm/README.rst [hunk 1]
```diff
@@ -78,6 +78,22 @@ The addresses of these devices are stored in a list.
 When a device does not respond within a certain period of time, it is removed from the list.
 Also the last measurement results with the peers are stored in this list.
 
+Energy consumption and the number of measurements performed in a given time depends on:
+
+   * Advertiser settings.
+   * Scanner settings.
+   * Extend of measurement window by ``extra_window_time_us`` in the ``dm_request`` structure.
```
[the remark this answers is not in the document]
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst]: This is additional time that extends the measurement window. That's why I changed it to: `Additional size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.`

File: samples/bluetooth/nrf_dm/README.rst [hunk 2]
```diff
@@ -78,6 +78,22 @@ The addresses of these devices are stored in a list.
 When a device does not respond within a certain period of time, it is removed from the list.
 Also the last measurement results with the peers are stored in this list.
 
+Energy consumption and the number of measurements performed in a given time depend on:
+
+   * Advertiser settings.
+   * Scanner settings.
+   * Additional size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:85]: How about "Extended size" instead of "Additional size"?

Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst]: Sounds good.

File: samples/bluetooth/nrf_dm/README.rst [hunk 3]
```diff
@@ -78,6 +78,22 @@ The addresses of these devices are stored in a list.
...
 
+Energy consumption and the number of measurements performed in a given time depend on:
+
+   * Advertiser settings.
+   * Scanner settings.
+   * Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
+
+The shorter the interval, the higher the frequency of broadcasting advertising packages.
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
+However, the higher number of packets transmitted also results in higher power consumption.
```
Comment (Tschet1) [inline samples/bluetooth/nrf_dm/README.rst:89]: maybe we can add a note here saying that this decreases the latency until a new device is picked up and that.
Additionally, maybe we can link the relevant KConfigs here.

File: samples/bluetooth/nrf_dm/README.rst [hunk 4]
```diff
@@ -78,6 +78,22 @@ The addresses of these devices are stored in a list.
...
+
+The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
+Like the short advertising interval, these values also have an impact on energy consumption.
+
+Extending the measurement window will increase the probability of measurement success for coarse synchronization.
+However, it affects the radio's turn-on time and thus, power consumption.
+
```
Comment (Tschet1) [inline samples/bluetooth/nrf_dm/README.rst:96]: Maybe we can also mention DM_MIN_TIME_BETWEEN_TIMESLOTS_US and DM_TIMESLOT_QUEUE_COUNT_SAME_PEER with which you can control how often the same peer is ranged.

**Review by Tschet1 — CHANGES_REQUESTED** [review]

File: samples/bluetooth/nrf_dm/README.rst (hunk 3 above)
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst:89]: Added additional note.

File: samples/bluetooth/nrf_dm/README.rst [hunk 5]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
 When a device does not respond within a certain period of time, it is removed from the list.
 Also the last measurement results with the peers are stored in this list.
 
+Energy consumption and the number of measurements performed in a given time depend on:
+
+   * Advertiser settings.
+   * Scanner settings.
+   * Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:85]:
```suggestion
Energy consumption and the number of measurements performed in a given time depend on the following factors:

* Advertiser settings.
* Scanner settings.
* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 6]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+
+The shorter the interval, the higher the frequency of broadcasting advertising packages.
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
+However, the higher number of packets transmitted also results in higher power consumption.
+
+The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
+Like the short advertising interval, these values also have an impact on energy consumption.
+
+How often the measurement will be executed is also managed by :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US`.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:94]:
```suggestion
How often the measurement will be executed is also managed by the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` Kconfig option.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 7]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+The :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` and :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_LENGTH` Kconfig options also indirectly affect this.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:95]:
```suggestion
The Kconfig options :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` and :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_LENGTH` also have an indirect impact on this.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 8]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+
+Extending the measurement window will increase the probability of measurement success for coarse synchronization.
+However, it affects the radio's turn-on time and thus, power consumption.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:98]:
```suggestion
However, it affects the turn-on time of the radio and thus, power consumption.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 9]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+
+At the time of application runtime, the additional delay of measurement execution is configured using ``start_delay_us`` in the :c:struct:`dm_request` structure.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:100]:
```suggestion
At the application runtime, the additional delay of measurement execution is configured using the ``start_delay_us`` parameter of the :c:struct:`dm_request` structure.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 10]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+This option helps adjust the synchronization.
+:kconfig:option:`CONFIG_DM_INITIATOR_DELAY_US` Kconfig options defines the initiator delay.
+Increasing these parameters increases the latency.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:103]:
```suggestion
This option helps in adjusting the synchronization.
The :kconfig:option:`CONFIG_DM_INITIATOR_DELAY_US` Kconfig option defines the initiator delay.
Increasing these parameters increases the latency.
```
Are the parameters increased or their values?

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/nrf_dm/README.rst (hunk 10 above)
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst]: It is about the value of these parameters.

**Review by peknis — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

File: samples/bluetooth/nrf_dm/README.rst [hunk 11]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
 When a device does not respond within a certain period of time, it is removed from the list.
 Also the last measurement results with the peers are stored in this list.
 
+Energy consumption and the number of measurements performed in a given time depend on the following factors:
+
+* Advertiser settings.
+* Scanner settings.
+* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
+
+The shorter the interval, the higher the frequency of broadcasting advertising packages.
```
Comment (Tschet1) [inline samples/bluetooth/nrf_dm/README.rst:87]:
```suggestion
The shorter the advertising interval, the higher the frequency at which the broadcaster sends out advertising packages.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 12]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
+However, the higher number of packets transmitted also results in higher power consumption.
+
+The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
+Like the short advertising interval, these values also have an impact on energy consumption.
```
Comment (Tschet1) [inline samples/bluetooth/nrf_dm/README.rst:92]:
```suggestion
Like the short advertising interval, these values also have an impact on power consumption and influence how fast and how often devices are ranged.
```
I think technically energy cannot be consumed.

File: samples/bluetooth/nrf_dm/README.rst [hunk 13]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
 Also the last measurement results with the peers are stored in this list.
 
+Energy consumption and the number of measurements performed in a given time depend on the following factors:
+
+* Advertiser settings.
+* Scanner settings.
+* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
+
+The shorter the interval, the higher the frequency of broadcasting advertising packages.
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
```
Comment (Tschet1) [inline samples/bluetooth/nrf_dm/README.rst:88]:
```suggestion
This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time and decreases the latency for the first measurement when the device is turned on or comes in reach.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 14]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
 When a device does not respond within a certain period of time, it is removed from the list.
 Also the last measurement results with the peers are stored in this list.
 
+Energy consumption and the number of measurements performed in a given time depend on the following factors:
```
Comment (Tschet1) [inline samples/bluetooth/nrf_dm/README.rst:81]: Maybe make this a separate section?

File: samples/bluetooth/nrf_dm/README.rst [hunk 15]
```diff
@@ -78,6 +78,30 @@ The addresses of these devices are stored in a list.
...
+* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
+
+The shorter the interval, the higher the frequency of broadcasting advertising packages.
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
+However, the higher number of packets transmitted also results in higher power consumption.
+
+The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
+Like the short advertising interval, these values also have an impact on energy consumption.
+
+How often the measurement will be executed is also managed by the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` Kconfig option.
```
[the remark this answers is not in the document]
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst:94]: If `CONFIG_DM_TIMESLOT_RESCHEDULE` is disabled then the next timeslot will not be allocated by the dm module. The next measurement will take after the next synchronization. So how often the device is measured depends on the frequency of synchronization. 
`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` affects this indirectly.

[the remark this answers is not in the document]
Comment (Tschet1) [reply samples/bluetooth/nrf_dm/README.rst:94]: Should we also add a note about that config?

File: samples/bluetooth/nrf_dm/README.rst (hunk 12 above)
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst]: Fixed.

File: samples/bluetooth/nrf_dm/README.rst (hunk 15 above)
[the remark this answers is not in the document]
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst]: Good point. Added.

**Review by Tschet1 — APPROVED** [review]

File: samples/bluetooth/nrf_dm/README.rst [hunk 16]
```diff
@@ -78,6 +78,35 @@ The addresses of these devices are stored in a list.
...
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time and decreases the latency for the first measurement when the device is turned on or comes in reach.
+However, the higher number of packets transmitted also results in higher power consumption.
+
+The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
+Like the short advertising interval, these values also have an impact on power consumption and influence how often devices are ranged.
+
+After the first measurement, the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` and the :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` Kconfig options define how fast the same device is measured again.
+The Kconfig options :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` and :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_LENGTH` define how many measurements are being scheduled.
+The more measurements that are being scheduled, the more measurements take place, which increases power consumption but gives more frequent rangings.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:99]:
```suggestion
The more measurements are scheduled, the more of them take place, which increases power consumption but gives more frequent rangings.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 17]
```diff
@@ -78,6 +78,35 @@ The addresses of these devices are stored in a list.
...
+If the :kconfig:option:`CONFIG_DM_TIMESLOT_RESCHEDULE` Kconfig option is disabled, the next timeslot for measurement execution will be allocated after the measurement request is added using the :c:func:`dm_request_add` function.
+
+Extending the measurement window will increase the probability of measurement success for coarse synchronization.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:102]:
```suggestion
Extending the measurement window increases the probability of measurement success for coarse synchronization.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 18]
```diff
@@ -78,6 +78,35 @@ The addresses of these devices are stored in a list.
...
+*****************
+
+Power consumption and the number of measurements performed in a given time depend on the following factors:
+
+* Advertiser settings.
+* Scanner settings.
+* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
+
+The shorter the advertising interval, the higher the frequency at which the broadcaster sends out advertising packages.
+This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time and decreases the latency for the first measurement when the device is turned on or comes in reach.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:91]:
```suggestion
This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
It also decreases the latency for the first measurement when the device is turned on or comes in reach.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 19]
```diff
@@ -78,6 +78,35 @@ The addresses of these devices are stored in a list.
...
+However, the higher number of packets transmitted also results in higher power consumption.
+
+The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
+Like the short advertising interval, these values also have an impact on power consumption and influence how often devices are ranged.
```
Comment (peknis) [inline samples/bluetooth/nrf_dm/README.rst:95]:
```suggestion
Like the short advertising interval, these values also have an impact on power consumption and the ranging frequency of devices.
```

File: samples/bluetooth/nrf_dm/README.rst [hunk 20]
```diff
@@ -78,6 +78,35 @@ The addresses of these devices are stored in a list.
...
+
+After the first measurement, the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` and the :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` Kconfig options define how fast the same device is measured again.
```
[the remark this answers is not in the document]
Comment (maje-emb) [reply samples/bluetooth/nrf_dm/README.rst]: Right, changed to _often_.

**Review by peknis — APPROVED** [review]

**Review by grochu — APPROVED** [review]
The description looks good.

**Pull request merged**